packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -23,113 +26,89 @@ import BlockUnknownConvertButton from './block-unknown-convert-button';
-								},
-								className: toggleClassname,
-								extraProps: { 'aria-expanded': isOpen },
-							} ] } />
-						);
-					} }
-					renderContent={ ( { onClose } ) => (
-						<NavigableMenu className="editor-block-settings-menu__content block-editor-block-settings-menu__content">
-							<__experimentalBlockSettingsMenuFirstItem.Slot fillProps={ { onClose } } />
-							{ count === 1 && (
-								<BlockUnknownConvertButton
-									clientId={ firstBlockClientId }
-								/>
-							) }
-							{ count === 1 && (
-								<BlockHTMLConvertButton
-									clientId={ firstBlockClientId }
+				<Toolbar>
+					<DropdownMenu
+						icon="ellipsis"
+						label={ __( 'Block options' ) }
+						position="bottom right"
+						className="block-editor-block-settings-menu"
+					>
+						{ ( { onClose } ) => (
+							<Fragment>
+								<__experimentalBlockSettingsMenuFirstItem.Slot
+									fillProps={ { onClose } }
 								/>
-							) }
-							{ ! isLocked && canDuplicate && (
-								<MenuItem
-									className="editor-block-settings-menu__control block-editor-block-settings-menu__control"
-									onClick={ onDuplicate }
-									icon="admin-page"
-									shortcut={ shortcuts.duplicate.display }
-								>
-									{ __( 'Duplicate' ) }
-								</MenuItem>
-							) }
-							{ ! isLocked && (
-								<Fragment>
+								{ count === 1 && (
+									<BlockUnknownConvertButton
+										clientId={ firstBlockClientId }
+									/>
+								) }
+								{ count === 1 && (
+									<BlockHTMLConvertButton
+										clientId={ firstBlockClientId }
+									/>
+								) }
+								{ ! isLocked && canDuplicate && (
 									<MenuItem
-										className="editor-block-settings-menu__control block-editor-block-settings-menu__control"
-										onClick={ onInsertBefore }
-										icon="insert-before"
-										shortcut={ shortcuts.insertBefore.display }
+										onClick={ onDuplicate }
+										icon="admin-page"
+										shortcut={ shortcuts.duplicate.display }
 									>
-										{ __( 'Insert Before' ) }
+										{ __( 'Duplicate' ) }
 									</MenuItem>
+								) }
+								{ ! isLocked && (
+									<Fragment>
+										<MenuItem
+											onClick={ onInsertBefore }
+											icon="insert-before"
+											shortcut={ shortcuts.insertBefore.display }
+										>
+											{ __( 'Insert Before' ) }
+										</MenuItem>
+										<MenuItem
+											onClick={ onInsertAfter }
+											icon="insert-after"
+											shortcut={ shortcuts.insertAfter.display }
+										>
+											{ __( 'Insert After' ) }
+										</MenuItem>
+									</Fragment>
+								) }
+								{ count === 1 && (
+									<BlockModeToggle
+										clientId={ firstBlockClientId }
+										onToggle={ onClose }
+									/>
+								) }
+								<__experimentalBlockSettingsMenuPluginsExtension.Slot
+									fillProps={ { clientIds, onClose } }
+								/>
+								<DropdownMenuSeparator />
```

Comment:
I have some concerns about this component which is 100% visual and won't be noticeable for those having sight issues. The question is whether this separator exists only for better aesthetics or should menu groups be used instead?

Reply:
> I have some concerns about this component which is 100% visual and won't be noticeable for those having sight issues. The question is whether this separator exists only for better aesthetics or should menu groups be used instead?

It's a very good point. Without digging into specifics, I would imagine you'd be correct in thinking that proper groupings would have preferable semantics than what can be communicated with this line. I could see an attempt at an argument that some lines are truly intended to be purely visual flourishes, though it seems a weak one and certainly one which should be limited, if not actively discouraged. At the very least, it's not clear to me why not to use a proper [`hr` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr) over the existing `div`.

We have the [`MenuGroup` component](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/menu-group) which might serve some help here, though (a) it seems to assume a label not currently present in the block settings menu and (b) [its implementation](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/menu-group/index.js) doesn't really seem to include anything particularly notable outside the labeling.

The more I look at [the menu entries](https://user-images.githubusercontent.com/1779930/55725309-693fa780-59db-11e9-833d-164b5094e7bc.png) and trying to consider what these groupings might be, the more I wonder if Remove Block really is in any way of a different classification than the other items, or if the line truly does just serve as visually emphasis (for its frequency? as an indicator of danger?).

Action items:

- Don't expose this as a publicly-available component
- Decide whether `MenuGroup` is possible, recommended (advice welcome)